### PR TITLE
ONLY FOR TESTING - DO NOT MERGE - ECDSA malleability tests for community 

### PR DESCRIPTION
**Pull request by joyqvq**

### CODE REVIEW

**Review by kchalkias — COMMENTED** [review]
did we test if it works in Sui with keytool (for some reason?), I can understand that fastcrypto seems fine. did we pin the latest fastcrypto version in Sui? @jonas-lj mentioned we plan to do anyway

File: fastcrypto/src/tests/secp256r1_tests.rs [hunk 1]
```diff
@@ -126,6 +127,35 @@ fn verify_valid_signature() {
...
+            let sig = kp.sign(&msg);
+            println!("is_high={:?}", sig.sig.s().is_high());
+            println!("sig={:?}", sig.sig);
+            let sig_low = sig.sig.normalize_s();
+            println!("sig low={:?}", sig_low); // this is none meaning it is already lower s
+            println!("res={:?}", kp.public().verify(&msg, &sig));
+        }
+
+        // input your privkey here
+        let kp = Secp256r1KeyPair::from(Secp256r1PrivateKey::from_bytes(&[]).unwrap());
```
Comment (joyqvq) [inline fastcrypto/src/tests/secp256r1_tests.rs:155]: i would be interested to see if your privkey is inserted here

File: fastcrypto/src/secp256r1/mod.rs [hunk 2]
```diff
@@ -338,6 +338,7 @@ impl Secp256r1KeyPair {
             .0;
 
         let sig_low = sig.normalize_s().unwrap_or(sig);
+        println!("here is high: {:?}", sig_low.s().is_high());
```
Comment (joyqvq) [inline fastcrypto/src/secp256r1/mod.rs:341]: and what does this say?

**Comment by joyqvq** [comment]
> did we test if it works in Sui with keytool (for some reason?), I can understand that fastcrypto seems fine. did we pin the latest fastcrypto version in Sui? @jonas-lj mentioned we plan to do anyway

sui delegate the entire call to fastcrypto for signing. 
```
    pub fn new_secure<T>(value: &IntentMessage<T>, secret: &dyn Signer<Signature>) -> Self
    where
        T: Serialize,
    {
        Signer::sign(
            secret,
            &bcs::to_bytes(&value).expect("Message serialization should not fail"),
        )
    }
```

i cannot think of a reason why it will mangle the s() bytes specifically at all...

**Comment by FrankC01** [comment]
@joyqvq 

I've been working with @kchalkias on getting intermittent failures using secp256r1
Here is the test keystring you can potentially incorporate... `AqPmvgC0d98+QsB9W2TumfztXyyxN3FxiTCBf3Fbd3Mg`

The error I'm getting: 
`Invalid user signature: InvalidSignature { error: "General cryptographic error: The s value of ECDSA signature must be low" }`

**Comment by joyqvq** [comment]
> @joyqvq
> 
> I've been working with @kchalkias on getting intermittent failures using secp256r1 Here is the test keystring you can potentially incorporate... `AqPmvgC0d98+QsB9W2TumfztXyyxN3FxiTCBf3Fbd3Mg`
> 
> The error I'm getting: `Invalid user signature: InvalidSignature { error: "General cryptographic error: The s value of ECDSA signature must be low" }`

i tried this key and the signature is indeed produced with lowered s, at least with our rust impl

File: fastcrypto/src/tests/secp256r1_tests.rs [hunk 3]
```diff
@@ -126,6 +128,41 @@ fn verify_valid_signature() {
...
+        //     println!("res={:?}", kp.public().verify(&msg, &sig));
+        // }
+
+        let c = "AqPmvgC0d98+QsB9W2TumfztXyyxN3FxiTCBf3Fbd3Mg";
+        let g = Base64::decode(c).unwrap();
+        println!("g={:?}", g.len());
+        // input your privkey here
+        let kp = Secp256r1KeyPair::from(Secp256r1PrivateKey::from_bytes(&g[1..]).unwrap());
+        let sig1 = kp.sign(&msg); // check if sig == sig1 
+        println!("sig={:?}", sig1.sig);
```
Comment (joyqvq) [inline fastcrypto/src/tests/secp256r1_tests.rs:161]: sig=ecdsa::Signature<NistP256>(FBCBDE65783C996D3D8959E7639C7315E82BC52B637860DA1C887E5F3F4A246D5E5309EB0A8588AF83080C3BFEEA275C07B6B230768AE2A05C1A6097107A7FB4)

this is indeed in lower s

**Comment by FrankC01** [comment]
>i tried this key and the signature is indeed produced with lowered s, at least with our rust impl. i can play around your python lib this weekend and let you know.

OK, I'm around so if you need anything else from me let me know!

**Review by kchalkias — COMMENTED** [review]
We can close this and do some extra tests in the related Python lib. Atm there is no indication our Rust tools fail.

**Comment by FrankC01** [comment]
> We can close this and do some extra tests in the related Python lib. Atm there is no indication our Rust tools fail.

@joyqvq @kchalkias 

I'm around again today should any exchange needed re: Python ECDSA

**Comment by FrankC01** [comment]
@joyqvq 
I have not been able to resolve. Have you had a chance to look at the python code or results?

**Comment by joyqvq** [comment]
> @joyqvq I have not been able to resolve. Have you had a chance to look at the python code or results?

sorry for the delay, testing your python library today and will reply here.

**Comment by FrankC01** [comment]
@joyqvq Thanks for the heads up... look forward to your observations.

**Comment by joyqvq** [comment]
> @joyqvq Thanks for the heads up... look forward to your observations.

is this the test to reproduce? ```
def test_secp256r1_signing() -> None:
    """Test signing."""
    edkp = keypair_from_keystring(TEST_SECP256R1_KEYSTRING)
    sig = edkp.new_sign_secure(PAY_FROM_SECP256R1)
    assert sig.signature == R1_SIG
```

**Comment by FrankC01** [comment]
@joyqvq 
Pretty much. I'm assuming that the `PAY_FROM_SECP256R1` is a intent prefixed byte string.

I'm also on 
discord: fastfrank#8904
telegram: @fastfrank

If you want to chat

**Pull request closed without merging**